File: C++/2447_별 찍기 - 10/main.cpp
#include <iostream>
using namespace std;

bool isStar(const int& x, const int& y, const int& entire) {
	if (x >= entire / 3 and x < entire / 3 * 2 and y >= entire / 3 and y < entire / 3 * 2) {
		return false;
	}
	else {
		if (entire == 3) {
			return true;
		}
		else {
			return isStar(x % (entire / 3), y % (entire / 3), entire / 3);
		}
	}
}

int main() {
	cin.tie(NULL); ios_base::sync_with_stdio(false);

	int N; cin >> N;

	for (int col{}; col < N; ++col) {
		for (int row{}; row < N; ++row) {
			if (isStar(row, col, N)) {
				cout << '*';
			}
			else {
				cout << ' ';
			}
		}

		cout << '\n';
	}

	return 0;
}

File: Python/2447_별 찍기 - 10/main.py
import sys

input = sys.stdin.readline


def isStar(row, col, scale):
    while True:
        scaleDiv3 = scale / 3

        if (
            row >= scaleDiv3
            and row < scaleDiv3 * 2
            and col >= scaleDiv3
            and col < scaleDiv3 * 2
        ):
            return False
        elif scale == 3:
            return True
        else:
            row %= scaleDiv3
            col %= scaleDiv3
            scale /= 3


a = int(input())

entireStr = ""
for col in range(a):
    for row in range(0, a, 3):
        if isStar(row, col, a):
            if col % 3 == 1:
                entireStr += "* *"
            else:
                entireStr += "***"
        else:
            entireStr += "   "
    entireStr += "\n"

print(entireStr)
